src/RecyclableMemoryStream.cs

```diff
@@ -699,7 +691,7 @@ public override byte[] ToArray()
             this.CheckDisposed();
 
             string stack = this.memoryManager.GenerateCallStacks ? Environment.StackTrace : null;
```

Comment:
Stack trace should be generated only if really needed.

Reply:
That's what it is doing? Are you saying that it should only record stack trace if ThrowExceptionOnToArray is also true? There is a valid scenario where I still want to allow it to go through, but want it logged to debug and remove.

Reply:
Unless I checked all code paths wrong, there's the chance a stack trace will be generated for nothing.